text/0000-cargo-libstd-awareness.md

```diff
@@ -0,0 +1,96 @@
+- Feature Name: cargo_libstd_awareness
+- Start Date: 2015-05-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Currently, all packages implicitly depend on libstd. This makes Cargo unsuitable for packages that
+need a custom-built libstd, or otherwise depend on crates with the same names as libstd and the
+crates behind the facade. The proposed fixes also open the door to a future where libstd can be
+Cargoized.
+
+# Motivation
+
+Bare-metal work cannot use a standard build of libstd. But since any crate built with Cargo can link
+with a system-installed libstd if the target matches, using Cargo for such projects can be irksome
+or impossible.
+
+Cargoizing libstd also generally simplifies the infrastructure, and makes cross compiling much
+slicker, but that is a separate discussion.
+
+Finally, I first raised this issue here: https://github.com/rust-lang/Cargo/issues/1096 Also, there
+are some (heavily bit-rotted) projects at https://github.com/RustOS-Fork-Holding-Ground that depend
+on each other in the way this RFC would make much more feasible.
+
+# Detailed design
+
+The current situation seems to be more of an accident of `rustc`'s pre-Cargo history than an
+explicit design decision. Cargo passes the location and name of all depended on crates to `rustc`.
+This is good because it means that that no undeclared dependencies on other Cargo packages can leak
+through. However, it also passes in `--sysroot /path/to/some/libdir`, the directory being were
+libstd is. This means packages are free to use libstd, the crates behind the facade, or none of the
+above, with Cargo being none the wiser.
+
+The only new interface proposed is a boolean field to the package meta telling Cargo that the
+package does not depend on libstd by default. This need not imply Rust's `no_std`, as one might want
+to `use` their own build of libstd by default. To disambiguate, this field is called
+q`no-implicit-deps`; please, go ahead and bikeshead the name. `no-implicit-deps` is false by
+default to maintain compatibility with existing packages.
+
+The meaning of this flag is defined in 3 phases, where each phase extends the last. The idea being
+is that while earlier phases are easier to implement, later phases yield a more elegant system.
+
+## Phase 1
+
+Add a `--no-sysroot` flag to `rustc`, and pass that to `rustc` is the case that `no-implicit-deps`
```

Comment:
Similar here; instead of `--no-sysroot`, make it `--use-sysroot=false` (with the default being `true`).